R/learner_abess_classif_abess.R

```diff
@@ -0,0 +1,149 @@
+#' @author abess-team
+#' @name mlr_learners_classif.abess
+#'
+#' @description
+#' Adaptive best-subset selection for classification.
+#' Calls [abess::abess()] from \CRANpkg{abess}.
+#'
+#' @templateVar id classif.abess
+#' @template learner
+#' @section Custom mlr3 defaults:
+#' * `num.threads`:
+#'   * Actual default: 0L
+#'   * Adjusted default: 1L
+#'   * Reason for change: avoid clashes between the mlr3 parallelization and the learner's internal parallelization.
+#'
+#' @template seealso_learner
+#' @template example
+#' @export
+LearnerClassifAbess = R6Class("LearnerClassifAbess",
+  inherit = LearnerClassif,
+  public = list(
+    #' @description
+    #' Creates a new instance of this [R6][R6::R6Class] class.
+    initialize = function() {
+      param_set = ps(
+        family = p_fct(
+          default = "binomial",
+          levels = c("binomial", "multinomial", "ordinal"),
+          tags = "train"
+        ),
+        tune.path = p_fct(
+          default = "sequence",
+          levels = c("sequence", "gsection"),
+          tags = "train"
+        ),
+        tune.type = p_fct(
+          default = "gic",
+          levels = c("gic", "aic", "bic", "ebic", "cv"),
+          tags = "train"
+        ),
+        normalize = p_int(default = NULL, special_vals = list(NULL), tags = "train"),
+        support.size = p_uty(default = NULL, tags = "train"),
+        c.max = p_int(default = 2, lower = 1, tags = "train"),
+        gs.range = p_uty(default = NULL, tags = "train"),
+        lambda = p_dbl(default = 0.0, lower = 0.0, tags = "train"),
+        always.include = p_uty(default = NULL, tags = "train"),
+        group.index = p_uty(default = NULL, tags = "train"),
+        init.active.set = p_uty(default = NULL, tags = "train"),
+        splicing.type = p_int(default = 2, lower = 1, upper = 2, tags = "train"),
+        max.splicing.iter = p_int(default = 20, lower = 1, tags = "train"),
+        screening.num = p_int(default = NULL, lower = 0, special_vals = list(NULL), tags = "train"),
+        important.search = p_int(default = NULL, lower = 0, special_vals = list(NULL), tags = "train"),
+        warm.start = p_lgl(default = TRUE, tags = "train"),
+        nfolds = p_int(default = 5,  tags = "train"),
+        foldid = p_uty(default = NULL, tags = "train"),
+        cov.update = p_lgl(default = FALSE, tags = "train"),
+        newton = p_fct(default = "exact", levels = c("exact", "approx"), tags = "train"),
+        newton.thresh = p_dbl(default = 1e-6, lower = 0.0, tags = "train"),
+        max.newton.iter = p_int(default = NULL, lower = 1, special_vals = list(NULL), tags = "train"),
+        early.stop = p_lgl(default = FALSE, tags = "train"),
+        ic.scale = p_dbl(default = 1.0, lower = 0.0, tags = "train"),
+        num.threads = p_int(default = 0L, lower = 0L, tags = c("train", "threads")),
+        seed = p_int(default = 1, tags = "train")
+      )
+      param_set$values$num.threads = 1L
```

Comment:
Either you add the documentation of the parameter in the section "Initial parameter values" or you do 
a `pv$num.threads = pv$num.threads %??% 1L` in the `.train()` call below. Default parameters are not the same as initial values. I prefer the second solution

Reply:
No, this only assigns 1 if the value of num.thread is NULL :)